vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -1160,34 +1563,29 @@ private void reset() {
                 "if($0.$connector) $0.$connector.reset();", getElement()));
     }
 
-    private DataCommunicator<T> getDataCommunicator() {
-        initDataCommunicator();
-        return dataCommunicator;
-    }
-
-    private void initDataCommunicator() {
-        if (dataCommunicatorInitializer != null) {
-            /*
-             * Init the Data Communicator: 1. Lazily, when the data lazy loading
-             * is used. Initialization occurs when the user clicks on the
-             * dropdown to view the list of items. 2. Eagerly, when the items
-             * are set explicitly or in-memory Data Provider is used.
-             */
-            dataCommunicatorInitializer.init();
-        } else if (dataCommunicator == null) {
-            /*
-             * If the user hasn't provided any data, initialize with empty data
-             * set.
-             */
-            setItems();
+    private void initDataCommunicatorIfEmpty() {
+        if (dataCommunicator == null) {
```

Comment:
Is there any point in delaying this instead of doing it directly in the constructor and removing this method and the places where it is called from (getting data view) ??? If there is, please answer as comment in code

Reply:
Oh, yes, you're right, no need to postpone it in the constructor. Will fix that

Reply:
Done